plugin/evm/validators_transitioner.go

```diff
@@ -0,0 +1,138 @@
+// (c) 2019-2021, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+
+	"github.com/flare-foundation/flare/ids"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// ValidatorsTransitioner transitions validators from a static set of validators
+// to a growing set of dynamic validators over a number of smooth steps.
+type ValidatorsTransitioner struct {
+	validators map[ids.ShortID]uint64
+	providers  ValidatorRetriever
+	cache      *lru.Cache
+}
+
+// NewValidatorsTransitioner creates a transition from the given default validators
+// to the validators retrieved from the given FTSO validators retriever.
+func NewValidatorsTransitioner(validators map[ids.ShortID]uint64, providers ValidatorRetriever, options ...CacheOption) *ValidatorsTransitioner {
+
+	cfg := DefaultCacheConfig
+	for _, opt := range options {
+		opt(&cfg)
+	}
+
+	cache, _ := lru.New(int(cfg.CacheSize))
+	v := ValidatorsTransitioner{
+		validators: validators,
```

Comment:
Perhaps it's because I lacked some background on this, having this named `defaultValidators` would have clarified to me that these are, in fact, default validators, and will not, at any point, be some merged set of the original validators and the FTSO validators.

Reply:
I didn't want to use longer naming there, as `providers` is pretty clear I think, and `defaults` sounds dumb.